packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -260,12 +285,18 @@ abstract class WidgetsBinding extends BindingBase implements GestureBinding, Ren
   // When editing the above, also update rendering/binding.dart's copy.
   @override
   void beginFrame() {
-    assert(!_buildingDirtyElements);
-    _buildingDirtyElements = true;
-    buildOwner.buildDirtyElements();
-    _buildingDirtyElements = false;
+    assert(!debugBuildingDirtyElements);
+    assert(() {
+      debugBuildingDirtyElements = true;
+      return true;
+    });
+    buildOwner.buildScope(renderViewElement);
     super.beginFrame();
     buildOwner.finalizeTree();
+    assert(() {
```

Comment:
If the code above throws and the `--hot` app is restarted, will this flag be reset?